### DISCUSSION TITLE
Wiznet5k on STM32

### DISCUSSION THREAD

**Discussion by victorallume** [micropython/micropython#10796] [Hardware & Peripherals]
I'm trying to get a Wiznet 5k module running with micropython on an STM32H7 (specifically, the Arduino Portenta H7). If I add `MICROPY_PY_NETWORK_WIZNET5K=5500` to the boards `mpconfigboard.mk` it compiles fine, but after flashing it to the board I get a `FATAL ERROR: MemManage` error (on the UART; the USB VCP doesn't come up). Any pointers on how to get this working? (I know the Portenta has an on-board ETH MAC, but I need to use the Wiznet in this case)

**Comment by victorallume** [comment]
A bit more info; I've manage to successfully compile and run micropython with the Wiznet library for the PYBoard 1.1; the build system seems to use the same directive (MICROPY_PY_NETWORK_WIZNET5K). I've tried it on the PyBoard with both LWIP enabled and disabled, and both seem to compile and run happily.

**Comment by victorallume** [comment]
Have made some progress; the device was running out of RAM. There is only 512KB on the MCU, which is all that micropython is using (the 8MB SDRAM doesn't appear to be utilised). When I disabled a bunch of other functions (Bluetooth, SSL, Wifi), it was able to compile and run it without the MemManage error, but this still occurs on boot when I compile with LWIP. When I compile without LWIP, I can boot and set up a WIZNET5K object, but as soon as I try `nic.isconnected()` the MemManage error occurs. `gc.mem_free()` immediately before calling `isconnected()` shows a large amount of free RAM (almost 500k)

**Reply by dpgeorge** [reply]
The MemManage error is a low-level ARM exception, possibly due to the MPU (memory protection unit) catching an invalid read (or write) of memory.  This could be for example an out-of-bounds memory access, like a seg fault.  It doesn't have anything to do with the MicroPython GC heap.

You should try to trace the error with a debugger, see what memory location it's faulting on.
